### PR TITLE
Keycloak operator 8.0.1

### PR DESCRIPTION
**Pull request by pb82**

### CODE REVIEW

File: pkg/products/rhsso/reconciler.go [hunk 1]
```diff
@@ -215,105 +316,217 @@ func (r *Reconciler) reconcileTemplates(ctx context.Context, installation *integ
 	return integreatlyv1alpha1.PhaseCompleted, nil
 }
 
+// workaround: the keycloak operator creates a route with TLS passthrough config
+// this should use the same valid certs as the cluster itself but for some reason the
+// signing operator gives out self signed certs
+// to circumvent this we create another keycloak route with edge termination
```
Comment (philbrookes) [inline pkg/products/rhsso/reconciler.go]: Is this going to be addressed in a future release of keycloak / the keycloak-operator? If so, can we create a Jira to remove this workaround at that point, so we don't forget?

File: pkg/products/rhsso/reconciler.go [hunk 2]
```diff
@@ -366,35 +572,39 @@ func (r *Reconciler) setupOpenshiftIDP(ctx context.Context, installation *integr
...
-			r.Config.GetHost() + "/auth/realms/openshift/broker/openshift-v4/endpoint",
-		},
-		GrantMethod: oauthv1.GrantHandlerPrompt,
+		Secret:       clientSecret,
+		RedirectURIs: redirectUris,
+		GrantMethod:  oauthv1.GrantHandlerPrompt,
 	}
 
-	_, err = r.ReconcileOauthClient(ctx, installation, oauthc, serverClient)
+	_, err = r.ReconcileOauthClient(ctx, inst, oauthClient, serverClient)
```
Comment (philbrookes) [inline pkg/products/rhsso/reconciler.go]: We refer to the installation object as `installation` everywhere now, to try to keep our code consistent

File: pkg/products/rhsso/reconciler.go [hunk 3]
```diff
@@ -494,15 +702,15 @@ func getUserDiff(keycloakUsers []*aerogearv1.KeycloakUser, openshiftUsers []user
 
 	var deleted []int
 	for i, kcUser := range keycloakUsers {
-		if kcUser.UserName != CustomerAdminUser.UserName && !OsUserInKc(openshiftUsers, kcUser) {
```
Comment (philbrookes) [inline pkg/products/rhsso/reconciler.go]: is this a typo? `OsUserInKc` => `IsUserInKc`?

Comment (pb82) [reply pkg/products/rhsso/reconciler.go]: I think this stands for `Openshift user in Keycloak`

File: pkg/products/rhsso/reconciler.go (hunk 2 above)
Comment (pb82) [reply pkg/products/rhsso/reconciler.go]: will update

File: pkg/products/rhsso/reconciler.go (hunk 1 above)
Comment (pb82) [reply pkg/products/rhsso/reconciler.go]: Yes, the PR to update the route in the keycloak operator has now been merged, but this is not in the 8.0.1 release: https://github.com/keycloak/keycloak-operator/pull/115

**Pull request merged**